src/bun.js/test/jest.zig

```diff
@@ -2009,6 +2009,70 @@ pub const Expect = struct {
         return .zero;
     }
 
+    pub fn toBeInstanceOf(this: *Expect, globalObject: *JSC.JSGlobalObject, callFrame: *JSC.CallFrame) callconv(.C) JSValue {
+        defer this.postMatch(globalObject);
+
+        const thisValue = callFrame.this();
+        const _arguments = callFrame.arguments(1);
+        const arguments: []const JSValue = _arguments.ptr[0.._arguments.len];
+
+        if (arguments.len < 1) {
+            globalObject.throwInvalidArguments("toBeInstanceOf() requires 1 argument", .{});
+            return .zero;
+        }
+
+        if (this.scope.tests.items.len <= this.test_id) {
+            globalObject.throw("toBeInstanceOf() must be called in a test", .{});
+            return .zero;
+        }
+
+        active_test_expectation_counter.actual += 1;
+
+        const expected_value = arguments[0];
+        if (!expected_value.jsType().isFunction()) {
```

Comment:
Can you either add a TODO here to check that the function is a constructor or add a binding for `isConstructor`? 

Checking that the JSType is one of the function types isn't precise enough. Functions can callable and/or constructable. The JSType can be whatever we want, though in most cases it will be one of the 3 that JSType currently checks (just not every case, in theory)

```suggestion
        if (!expected_value.isConstructor()) {
```

This `isConstructor` function doesn't exist, but it would be nearly identical to `isCallable`:

https://github.com/oven-sh/bun/blob/4c3879814289f79d81aef30e55c4bad9145c2af8/src/bun.js/bindings/bindings.zig#L3528-L3530

https://github.com/oven-sh/bun/blob/4c3879814289f79d81aef30e55c4bad9145c2af8/src/bun.js/bindings/bindings.cpp#L2425-L2428